### PR TITLE
docs(matrices): document polynomial inertia cells

### LINKED ISSUE #3456: [Matrix analysis] Return exact inertia cells for one-parameter symmetric polynomial matrices
## Missing postcondition

For a symmetric univariate polynomial matrix A(t) over QQ and a closed rational interval [a,b], return a finite exact partition into point cells and interval cells on which inertia (n_positive,n_negative,n_zero) is constant, including every singular specialization.

This is a proposed atomic parameter profile, not a sampled eigenvalue plot or a search for a favorable parameter.

## Existing capabilities and distinction

Existing fixed-matrix inertia and symbolic polynomial operations can support an implementation, but none of the inspected manifests exposes the complete parameter partition. #1771 owns rational congruence diagonalization of a fixed form; #3283 concerns symbolic-matrix carriers. Neither supplies the parameter-wide inertia result.

The material computation is isolating all transition values, treating singular fibers exactly, and certifying that no transition is missed. Returning inertia at several rational samples is insufficient.

## Contract and representation

- Input: one ordered-axis square symmetric matrix with QQ[t] entries, named parameter, rational interval endpoints.
- Output: source-bound ordered cells covering exactly [a,b], endpoint inclusion retained, exact inertia per cell.
- Nonrational endpoints need canonical real-algebraic values with isolating information and the intended real embedding. Reuse existing scalar and interval carriers; do not serialize approximations as exact boundaries.
- Define a deterministic partition convention. A valid refinement is mathematically sufficient; canonical merging of adjacent equal-inertia strata can be a native normalization rather than a new operation.

Constant matrices, the zero matrix, zero-dimensional shapes if supported by the canonical carrier, and singleton parameter intervals require explicit semantics.

## Motivating fixtures

A(t)=diag(t^2-2,0) on [-2,2] has inertia:

- (1,0,1) for |t|>sqrt(2);
- (0,1,1) for |t|<sqrt(2);
- (0,0,2) at t=±sqrt(2).

Its determinant is identically zero, so partitioning solely at determinant roots fails completely. The rational sample values were checked exactly; the displayed complete partition follows directly from the diagonal entries.

Also test diag(t^2,1): the sign does not flip at zero, but nullity changes there. Test [[t,1],[1,-t]], whose inertia is constantly (1,1,0), and matrices with repeated algebraic transition roots.

## Algorithm and admission questions

A bounded exact route can use characteristic-polynomial coefficients, generic rank, and real-root isolation. For a symmetric matrix of generic rank r, the coefficient equal to the product of its r nonzero eigenvalues is a nonzero polynomial; its zeros contain rank-drop parameters. Establish this reduction and handle the identically zero matrix separately. Nonzero eigenvalues cannot change sign on a rank-constant connected cell.

At isolated algebraic points, use exact specialization and sign/inertia arithmetic, not floating eigenvalues. Candidate root isolation plus fixed-fiber inertia is an implementation, not a runtime proof narration.

Preflight matrix order, entry degree/support/height, determinant or characteristic-coefficient intermediate growth, root-isolation work, algebraic specialization, cell count, and result size. Evaluate maintained polynomial and real-algebraic backends before writing a custom isolator. No performance envelope is established by this proposal; benchmark nonconstant singular families before publication.

## Acceptance

Exact coverage, disjoint cell ownership of boundaries, inertia sums equal to matrix order, all rank-drop points retained, and agreement with independent exact specialization on bounded fixtures. Include persistent nullspaces, tangencies, endpoint singularities, and nonrational transition points.

Reject unsupported cost before execution; an isolation failure or timeout must not yield a supposedly complete partition. Keep native/MCP contracts identical.

### PR DESCRIPTION
**Pull request by morluto**
## Summary

Closes #3456.

The exact `matrix.polynomial.inertia_cells.compute` operation, native kernel,
canonical carriers, bounded admission, independent exact fixtures, and MCP
parity are already present on `main` through the merged polynomial-matrix
change. This PR completes the public operation reference surface by documenting
its exact alternating point/open partition, algebraic boundary representation,
degenerate cases, admission boundary, and native/MCP entry points, and links it
from the matrix-operation index.

## Validation

- `make docs-linkcheck`
- `make quick-scoped LANE=math TESTS=tests/math/matrices/inertia_cells/test_inertia_cells.py PATHS="src/jacobian/math/matrices/inertia_cells tests/math/matrices/inertia_cells tests/mcp/test_polynomial_inertia_cells.py docs/reference/operations/matrix"`
- `make quick-scoped LANE=catalog TESTS="tests/mcp/test_polynomial_inertia_cells.py tests/integration/catalog/test_public_operation_contract_conformance.py" PATHS="src/jacobian/math/matrices/inertia_cells tests/mcp/test_polynomial_inertia_cells.py docs/reference/operations/matrix"`
- `uv run pytest -q -o addopts='' -m exhaustive tests/integration/catalog/test_public_operation_contract_conformance.py -k matrix.polynomial.inertia_cells.compute`
- `make lint typecheck`

### CODE REVIEW

File: docs/reference/operations/matrix/polynomial-inertia-cells.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+isolating intervals. The partition is exact; it is not a sampled eigenvalue
+plot. In particular, `diag(t² - 2, 0)` on `[-2, 2]` retains both `±√2` even
+though its determinant is identically zero.
+
+The kernel first bounds the source, characteristic-polynomial growth,
+transition isolation, exact coefficient-sign work, sampling, and output size.
+It then processes disconnected support blocks independently, isolates all roots
+of the lowest nonzero characteristic coefficient, and uses exact specialization
+at point cells. The eigenvalue signs are obtained from the real-rooted
+characteristic polynomial's Descartes sign variations; no floating-point
```
[the remark this answers is not in the document]
Comment (morluto) [reply docs/reference/operations/matrix/polynomial-inertia-cells.md]: Fixed in 14eb41b6. The public reference now describes the stable exact partition and admission contract without publishing replaceable kernel strategy.
